io/jvm/src/main/scala/scalaz/effect/RTS.scala

```diff
@@ -463,7 +459,7 @@ private object RTS {
                       }
                     } else {
                       // Error caught:
-                      val value = -\/(error)
+                      val value = Left(error)
```

Comment:
Gotcha

Reply:
The weird thing is tho, why the tests are passing with such an horrible error in the implementation? Are we missing a case when the use of attempt should lead us to this line?

Reply:
The tests are passing because, coincidentally, you are using `err = Left(_)`, because this was the simplest way to get the tests compiling (e.g. `attemptEither` and `absolveEither`). If you change the tests to use a custom data type (e.g. `attemptMyEither`), or stop using the helpers and use `attempt` directly (but don't build `Either` in failure), you'll get class cast exceptions.

It's an unlucky confluence which makes the tests pass. 😄

Reply:
Oh, get it. I don't like that happening tho. But won't put too much effort on changing it if we are discussing about ditching `Disj`